### PR TITLE
Pretty Printing

### PR DESCRIPTION
**Pull request by gregoryyoung**
Pretty printing support for ScriptCS in REPL

(some of original changes are squashed in @glennblock commit)

```code
$ scriptcs
scriptcs (ctrl-c to exit or :help for help)
> Env.AddCustomPrinter<int>(x => "testing " + x)
Adding custom printer for Int32
> 5
testing 5
>  
```
For running scripts without repl you can also use:

> Env.Print(25)
testing 25

There is also an overload for Print< T >(T obj) that allows for Print<Base>(derived)

@glennblock and I were also discussing the possibility of extracting a Printers interface (a little odd in a test setup to pass the dependecy in).

As state before this is to open up conversation.

### CODE REVIEW

**Comment by glennblock** [comment]
Cool!

@filipw this doesn't move to console but it does introduce a Print method on Env which can be called outside the REPL.

**Comment by glennblock** [comment]
Let's get a unit of acceptance test

File: src/ScriptCs.Core/ScriptEnvironment.cs [hunk 1]
```diff
@@ -1,15 +1,38 @@
...
+
+        public void AddCustomPrinter<T>(Func<T, string> printer)
+        {
+            _console.WriteLine("Adding custom printer for " + typeof(T).Name);
+            _printers.AddCustomPrinter<T>(printer);
+        }
+
+        public void Print(object o)
+        {
+            Console.WriteLine(_printers.GetStringFor(o));
```
Comment (glennblock) [inline src/ScriptCs.Core/ScriptEnvironment.cs]: Shouldn't this be _console?

File: src/ScriptCs.Core/ScriptEnvironment.cs [hunk 2]
```diff
@@ -1,15 +1,38 @@
...
+        }
+
+        public void Print<T>(T o)
+        {
+            Console.WriteLine(_printers.GetStringFor<T>(o));
```
Comment (glennblock) [inline src/ScriptCs.Core/ScriptEnvironment.cs]: Same as above?

**Comment by glennblock** [comment]
Awesome work, thanks!

Make sure to add your name to our contributors page!

**Comment by filipw** [comment]
👍 thanks

**Pull request merged**